src/main/scala/com/fulcrumgenomics/commons/io/Io.scala

```diff
@@ -49,8 +50,28 @@ trait IoUtil {
 
   /** Creates a new InputStream to read from the supplied path. */
   def toInputStream(path: Path) : InputStream = {
-    val stream =  if (Files.isSameFile(path, Io.StdIn)) System.in else Files.newInputStream(path)
-    new BufferedInputStream(stream, bufferSize)
+    // Developer Note **IMPORTANT**:
+    // Do not wrap pipes in BufferedInputStream, as there is a java bug where seek() is called on the stream.  For
+    // example, we get this exception:
+    //   java.io.IOException: Illegal seek
+    //   at sun.nio.ch.FileDispatcherImpl.seek0(Native Method)
+    //   at sun.nio.ch.FileDispatcherImpl.seek(FileDispatcherImpl.java:76)
+    //   at sun.nio.ch.FileChannelImpl.position(FileChannelImpl.java:264)
+    //   at sun.nio.ch.ChannelInputStream.available(ChannelInputStream.java:116)
+    //   at java.io.BufferedInputStream.read(BufferedInputStream.java:353)
+    //   at sun.nio.cs.StreamDecoder.readBytes(StreamDecoder.java:284)
+    //   at sun.nio.cs.StreamDecoder.implRead(StreamDecoder.java:326)
+    //   at sun.nio.cs.StreamDecoder.read(StreamDecoder.java:178)
+    //   at java.io.InputStreamReader.read(InputStreamReader.java:184)
+    //   at java.io.BufferedReader.fill(BufferedReader.java:161)
+    // This is a known issue; see: https://github.com/samtools/htsjdk/issues/1084
+    if (Files.isSameFile(path, Io.StdIn)) System.in
+    else {
+      val stream = Files.newInputStream(path)
+      val attrs  = Files.readAttributes(path, classOf[BasicFileAttributes])
+      if (!attrs.isRegularFile && !attrs.isSymbolicLink && attrs.isOther) stream
```

Comment:
Would it make sense to resolve the Path fully using `Path.toRealPath()` first?  And then we can just worry about what the target is and not whether we were handed a symlink?  I think specifically I would suggest changing line 71 to be `val attrs = Files.readAttributes(path.toRealPath(), ...` so that if there is any issue opening the path on line 70 that the user provided path is reported in the exception.

Reply:
I went with this